File: src/main/java/com/footverse/order/dto/OrderStatusCountResponse.java
package com.footverse.order.dto;

import com.footverse.order.entity.OrderStatus;

/**
 * One row of the ADMIN dashboard's per-status order count (dto-spec §15, Sprint 13). The dashboard
 * always returns exactly one row per {@link OrderStatus} value, zero-filled by the service for a
 * status with no orders — the repository aggregate returns rows only for statuses that occur.
 *
 * @param status the status this row counts
 * @param count  the number of orders currently at this status
 */
public record OrderStatusCountResponse(OrderStatus status, long count) {
}

File: admin/src/features/dashboard/models/order-status-count-response.ts
/**
 * Lifecycle status of an order (dto-spec §4). Mirrors the frozen enum values —
 * the same five values as the `order` feature's own `OrderStatus`, defined
 * again here rather than imported: `AdminOrderSummaryResponse` is the one
 * cross-feature import this feature makes (sprint-13-plan Task 02 Design
 * Note), and a five-value string-literal union frozen since Sprint 1 carries
 * no meaningful drift risk the way a multi-field interface would.
 */
export type OrderStatus = 'PENDING' | 'CONFIRMED' | 'SHIPPING' | 'DELIVERED' | 'CANCELLED';

/**
 * One row of the ADMIN dashboard's per-status order count (dto-spec §15,
 * Sprint 13). The dashboard always returns exactly one row per `OrderStatus`
 * value, zero-filled server-side for a status with no orders. Mirrors the
 * frozen DTO field-for-field.
 */
export interface OrderStatusCountResponse {
  readonly status: OrderStatus;
  readonly count: number;
}
